modules/Emails/include/ListView/ListViewDataEmailsSearchOnCrm.php

```diff
@@ -0,0 +1,316 @@
+ * this program; if not, see http://www.gnu.org/licenses or write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
+
+include_once 'modules/Emails/include/ListView/ListViewDataEmailsSearchAbstract.php';
+
+class ListViewDataEmailsSearchOnCrm extends ListViewDataEmailsSearchAbstract {
+
+    /**
+     * @param array $filterFields
+     * @param array $request $_REQUEST
+     * @param string $where
+     * @param InboundEmail $inboundEmail
+     * @param array $params
+     * @param Email $seed
+     * @param bool $singleSelect
+     * @param ??? $id
+     * @param int $limit
+     * @param User $currentUser
+     * @param string $idField
+     * @param int $offset
+     * @return array
+     */
+    public function search($filterFields, $request, $where, InboundEmail $inboundEmail, $params, Email $seed, $singleSelect, $id, $limit, User $currentUser, $idField, $offset) {
+        // Fix fields in filter fields
+
+        $filterFields = $this->lvde->fixFieldsInFilter($filterFields, $request, $where);
+
+
+        // Filter imported emails based on the UID of the results from the IMap server
+
+        if(!empty($where)) {
+            $where .= ' AND ';
+        }
+        $crmWhere = $where . 'mailbox_id LIKE ' .'"' . $inboundEmail->id . '"';
+
+
+        // Populates CRM fields
+
+        $crmQueryArray = $this->lvde->getCrmQueryArray($crmWhere, $filterFields, $params, $seed, $singleSelect);
+
+
+        // get crm emails query
+
+        $crmEmailsQuery = $this->lvde->getCrmEmailsQuery($crmQueryArray, $params);
+
+
+
+
+        $this->lvde->setVariableName($seed->object_name, $crmWhere, $this->lvde->listviewName, $id);
+
+        SugarVCR::erase($seed->module_dir);
+        $this->lvde->seed =& $seed;
+        $totalCounted = empty($GLOBALS['sugar_config']['disable_count_query']);
+        $_SESSION['MAILMERGE_MODULE_FROM_LISTVIEW'] = $seed->module_dir;
+        if(empty($request['action']) || $request['action'] != 'Popup'){
```

Comment:
(class ListViewDataEmailsSearchOnIMap and class ListViewDataEmailsSearchOnCrm not completely refactored yet)